src/test/java/vn/zalopay/benchmark/core/sampler/GRPCSamplerGuiTest.java

```diff
@@ -219,4 +224,54 @@ public void testJSONMetadataGrpcTestElement() throws InterruptedException {
         Assert.assertNotNull(grpRequestPluginGUI);
         frame.dispose();
     }
+
+    @Test
+    public void testExceptionRequestWithListenerResultTree() throws Exception {
+        // init window
+        RenderAsText samplerResultTab = new RenderAsText();
+        ViewResultsFullVisualizerGui viewResultsFullVisualizerGui = new ViewResultsFullVisualizerGui(samplerResultTab);
+        JFrame frame = new JFrame("testExceptionRequestListenerResultTree");
+        frame.setPreferredSize(new Dimension(1024, 768));
+        Container contentPane = frame.getContentPane();
+        contentPane.add(viewResultsFullVisualizerGui, BorderLayout.CENTER);
+        frame.pack();
+        frame.setVisible(true);
+
+        // exec request
+        HostAndPort hostAndPort = HostAndPort.fromString(HOST_PORT);
+        GRPCSampler grpcSampler = new GRPCSampler();
+        grpcSampler.setName("testExceptionRequestListenerResultTree");
+        grpcSampler.setComment("dummyComment");
+        grpcSampler.setProtoFolder(PROTO_PATH_WITH_ERROR_IMPORT_FOLDER.toString());
+        grpcSampler.setLibFolder(LIB_FOLDER.toString());
+        grpcSampler.setMetadata(METADATA);
+        grpcSampler.setHost(hostAndPort.getHost());
+        grpcSampler.setPort(Integer.toString(hostAndPort.getPort()));
+        grpcSampler.setFullMethod(FULL_METHOD_INVALID);
+        grpcSampler.setDeadline("2000");
+        grpcSampler.setTls(false);
+        grpcSampler.setTlsDisableVerification(false);
+        grpcSampler.setChannelShutdownAwaitTime("5000");
+        grpcSampler.setRequestJson(REQUEST_JSON);
+        grpcSampler.threadStarted();
+        SampleResult sampleResult = grpcSampler.sample(null);
+
+        // set result
+        viewResultsFullVisualizerGui.add(sampleResult);
+        Assert.assertEquals(sampleResult.getResponseCode(), " 400");
+        Assert.assertEquals(sampleResult.getResponseMessage(), GrpcSamplerConstant.CLIENT_EXCEPTION_MSG);
+        Assert.assertTrue(new String(sampleResult.getResponseData()).contains("invalid.proto:11:1: File recursively imports itself: invalid.proto -> invalid.proto"));
+
+        // Destruction of the view
+        frame.dispose();
+
+        // Leave the view and wait indefinitely so that the JVM does not exit and the results are verified manually
```

Comment:
Could you please remove this comment code

Reply:
Hello, this is the code snippet that needs to be kept, you need to enable a different code snippet when you need to observe GUI styles.

I considered providing a `.jmx` file to look at GUI styles, but ran into problems with proto file paths and gave up, writing junit use cases instead

Let me optimize the comment

Reply:
- Unless it can be run in our Gitlab pipeline, people rarely uncomment that code to run that tests in their local machine
- The GUI style should include in our unit tests for sampler GUI. The AbstractVisualizer should not include in our test since the JMeter will notify any plugin such as BackendListener, ActionListener, etc. the SamplerResult, and we can make sure our sampler result after completing gRPC request has a valid data
- Almost the GUI view is only used for debugging, and we can create a dummy AbstractVisualizer and verify the SamplerResult receives expected data from add(final SampleResult sample) methods is correctly

Reply:
- Hello, the GUI test case runs successfully, my last submission, nothing is done, but the test cases all succeed again. I wonder if it is because of the submission of `.codecov.yml` files that Gitlab Pipeline runs successfully.
- Yes, so I provided two use cases：
  - A is located in the`GrpcSamplerTest.testCanSendSampleRequestWithErrorProtoFolder()` Used to confirm SampleResult
  - A is located in the`GRPCSamplerGuiTest.testExceptionRequestWithListenerResultTree()`  For viewing styles

Reply:
Lets me merge this PR and next PR we should remove the comment code below, since, we don't use that code when running with Github action and rarely people will need to run it in locally

```java       
// GUI observation: Leave the view and wait indefinitely so that the JVM does not exit and the results are verified manually
//        Object lock = new Object();
//        while (true) {
//            synchronized (lock) {
//                lock.wait();
//            }
//        }```